crates/loader/src/local/assets.rs

```diff
@@ -20,22 +20,49 @@ use super::config::{RawDirectoryPlacement, RawFileMount};
 pub(crate) async fn prepare_component(
     raw_mounts: &[RawFileMount],
     src: impl AsRef<Path>,
-    base_dst: impl AsRef<Path>,
+    base_dst: Option<impl AsRef<Path>>,
     id: &str,
     exclude_files: &[String],
 ) -> Result<Vec<DirectoryMount>> {
-    tracing::info!(
-        "Mounting files from '{}' to '{}'",
-        src.as_ref().display(),
-        base_dst.as_ref().display()
-    );
-
-    let files = collect(raw_mounts, exclude_files, src)?;
-    let host = create_dir(&base_dst, id).await?;
-    let guest = "/".to_string();
-    copy_all(&files, &host).await?;
+    if let Some(base_dst) = base_dst {
+        tracing::info!(
+            "Mounting files from '{}' to '{}'",
+            src.as_ref().display(),
+            base_dst.as_ref().display()
+        );
 
-    Ok(vec![DirectoryMount { guest, host }])
+        let files = collect(raw_mounts, exclude_files, src)?;
+        let host = create_dir(&base_dst, id).await?;
+        let guest = "/".to_string();
+        copy_all(&files, &host).await?;
+
+        Ok(vec![DirectoryMount { guest, host }])
+    } else {
+        tracing::info!("directly mounting local asset directories into guest");
+
+        if !exclude_files.is_empty() {
+            bail!("exclusions not permitted when mounting directories directly")
```

Comment:
I'd flip this error around - if there are exclusions, it's because that's appropriate for the application, and it should be `--direct-mount` that yields, not the exclusions.  E.g. "this component cannot be run with --direct-mount because some files are excluded from mounting" or something like that?